torch/csrc/Exceptions.h

```diff
@@ -52,58 +56,118 @@ static inline void PyErr_SetString(PyObject* type, const std::string& message) {
-        ? e.what()                                                           \
-        : e.what_without_backtrace();                                        \
-    PyErr_SetString(PyExc_ValueError, torch::processErrorMsg(msg));          \
-    retstmnt;                                                                \
-  }                                                                          \
-  catch (const c10::TypeError& e) {                                          \
-    auto msg = torch::get_cpp_stacktraces_enabled()                          \
-        ? e.what()                                                           \
-        : e.what_without_backtrace();                                        \
-    PyErr_SetString(PyExc_TypeError, torch::processErrorMsg(msg));           \
-    retstmnt;                                                                \
-  }                                                                          \
-  catch (const c10::NotImplementedError& e) {                                \
-    auto msg = torch::get_cpp_stacktraces_enabled()                          \
-        ? e.what()                                                           \
-        : e.what_without_backtrace();                                        \
-    PyErr_SetString(PyExc_NotImplementedError, torch::processErrorMsg(msg)); \
-    retstmnt;                                                                \
-  }                                                                          \
-  catch (const c10::LinAlgError& e) {                                        \
-    auto msg = torch::get_cpp_stacktraces_enabled()                          \
-        ? e.what()                                                           \
-        : e.what_without_backtrace();                                        \
-    PyErr_SetString(THPException_LinAlgError, torch::processErrorMsg(msg));  \
-    retstmnt;                                                                \
-  }                                                                          \
-  catch (const c10::Error& e) {                                              \
-    auto msg = torch::get_cpp_stacktraces_enabled()                          \
-        ? e.what()                                                           \
-        : e.what_without_backtrace();                                        \
-    PyErr_SetString(PyExc_RuntimeError, torch::processErrorMsg(msg));        \
-    retstmnt;                                                                \
-  }                                                                          \
-  catch (torch::PyTorchError & e) {                                          \
-    auto msg = torch::processErrorMsg(e.what());                             \
-    PyErr_SetString(e.python_type(), msg);                                   \
-    retstmnt;                                                                \
-  }
+#if defined(USE_CUDA)
```

Comment:
I hate this. Can you instead move CUDAOutOfMemoryError so it is defined unconditionally and then eliminate the #if

Reply:
Sure, it seems that we'd lose the C10_CUDA_API declaration of the error though, and instead just have it as a C10_API,  not sure what the difference is there.

Reply:
it doesn't matter; the API decl has to match what library it's in but if you move it to libc10.so that's correct